lib/src/config.rs

```diff
@@ -215,6 +217,28 @@ pub struct RepoEncryption {
     pub nonce: secretbox::Nonce,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+/// Chunking Algorithms supported by rdedup
+pub enum ChunkingAlogrithm {
+    Bup,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+/// RepoChunking is the configuration of the chunking algorithm and parameters for the repo
+pub struct RepoChunking {
+    pub algorithm: ChunkingAlogrithm,
+    pub size: u32, // size is generic enough to work with different algorithms
```

Comment:
Don't worry I am using rusts powerful enums in a new commit. I seem to forget all the advantages of rust when I switch from my main work language Go to working on rust projects =(.

I guess I need to use it more at work =D